Automattic/wp-calypso
# Plans: Align Stats card with Add-Ons card for various prices

Related to https://github.com/Automattic/wp-calypso/issues/79867#issuecomment-1652335221

## Proposed Changes

* Refactor Stats indirect checkout products constants.
* Add support to `displayPriceText` for showing the price text directly.
* Use `jetpack_stats_pwyw_yearly` as the default Stats product on the plans page.

## Testing Instructions



* Spin up the change with the Calypso Live link.
* Navigate to `/plans/{jetpack-site-slug}?flags=stats/paid-stats`.
* Ensure the Stats card shows the price text `Varies` for getting the purchase.

|Before|After|
|-|-|
|<img width="1264" alt="截圖 2023-07-29 上午1 29 31" src="https://github.com/Automattic/wp-calypso/assets/6869813/3fade2fe-ba17-42b3-86d7-88ac142ac70b">|<img width="1268" alt="截圖 2023-07-29 上午1 16 41" src="https://github.com/Automattic/wp-calypso/assets/6869813/b80fa84a-1618-4344-9a0f-dede39828815">|

* Ensure the purchased Stats cards show respective product names accurately.

<img width="1269" alt="截圖 2023-07-29 上午1 33 11" src="https://github.com/Automattic/wp-calypso/assets/6869813/ed3e8028-2e6c-44a4-bf52-9a98cf47ed56">

## Pre-merge Checklist
